packages/core/src/components/popover-next/popoverTarget.tsx

```diff
@@ -0,0 +1,152 @@
+        ? {
+              onBlur: handleTargetBlur,
+              onContextMenu: handleTargetContextMenu,
+              onFocus: handleTargetFocus,
+              onMouseEnter: handleMouseEnter,
+              onMouseLeave: handleMouseLeave,
+          }
+        : {};
+
+    // Ensure target is focusable if relevant prop enabled
+    const targetTabIndex = !isContentEmpty && !disabled && openOnTargetFocus && isHoverInteractionKind ? 0 : undefined;
+
+    const ownTargetProps = {
+        // N.B. this.props.className is passed along to renderTarget even though the user would have access to it.
+        // If, instead, renderTarget is undefined and the target is provided as a child, props.className is
+        // applied to the generated target wrapper element.
+        className: classNames(className, Classes.POPOVER_TARGET, {
+            [Classes.POPOVER_OPEN]: isOpen,
+            // this class is mainly useful for button targets
+            [Classes.ACTIVE]: isOpen && !isControlled && !isHoverInteractionKind,
+        }),
+        ref,
+        ...targetEventHandlers,
+    } satisfies React.HTMLProps<HTMLElement>;
+    const childTargetProps = {
+        "aria-expanded": isHoverInteractionKind ? undefined : isOpen,
+        "aria-haspopup": interactionKind === PopoverInteractionKind.HOVER_TARGET_ONLY ? undefined : popupKind ?? "menu",
+    } satisfies React.HTMLProps<HTMLElement>;
+
+    const targetModifierClasses = {
+        // this class is mainly useful for Blueprint <Button> targets; we should only apply it for
+        // uncontrolled popovers when they are opened by a user interaction
+        [Classes.ACTIVE]: isOpen && !isControlled && !isHoverInteractionKind,
+        // similarly, this class is mainly useful for targets like <Button>, <InputGroup>, etc.
+        [Classes.FILL]: fill,
+    };
+
+    let target: React.JSX.Element | undefined;
+
+    if (renderTarget !== undefined) {
+        const floatingProps = floatingData.getReferenceProps();
+
+        target = renderTarget({
+            ...ownTargetProps,
+            ...childTargetProps,
+            // Apply Floating UI's interaction props for renderTarget case
+            ...floatingProps,
+            className: classNames(ownTargetProps.className, targetModifierClasses),
+            // if the consumer renders a tooltip target, it's their responsibility to disable that tooltip
+            // when *this* popover is open
+            isOpen,
+            tabIndex: targetTabIndex,
+        });
+    } else {
+        const childTarget = Utils.ensureElement(Children.toArray(children)[0]);
+
+        if (childTarget === undefined) {
+            return null;
+        }
+
+        const clonedTarget = cloneElement(childTarget, {
+            ...childTargetProps,
+            className: classNames(childTarget.props.className, targetModifierClasses),
+            disabled: isOpen && isTooltipElement(childTarget) ? true : childTarget.props.disabled,
```

Comment:
Let's just use a regular boolean expression instead of a ternary.
```suggestion
            disabled: (isOpen && isTooltipElement(childTarget)) || childTarget.props.disabled,
```

Reply:
bef744acd3ada851303151d9a9e5aebb9f76ab6c